lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -607,6 +608,28 @@ class DocumentImpl extends NodeImpl {
     });
   }
 
+  // https://dom.spec.whatwg.org/#dom-document-createcdatasection
+  createCDATASection(data) {
+    if (data.indexOf("[[>") !== -1) {
+      throw new DOMException(DOMException.INVALID_CHARACTER_ERR,
+        "CDATA section data cannot contain the string \"]]>\"");
+    }
+
+//  if (HTML document) {
```

Comment:
@domenic this is the area i'd like to know the proper convention of being an "html document" so we can throw a `NotSupportedError`